src/model/consent/consent.ts

```diff
@@ -57,16 +66,20 @@ export class ConsentDB {
   }
 
   // Update Consent credential
+  // Only non-null Consent fields are updated
   public updateCredentials (consent: Consent): Promise<Consent[]> {
+    const validatedConsent = {}
+
+    for (const key of Object.keys(consent)) {
+      // Only allow nullProtectedFields with non-null values for update
+      if (this.nullProtectedProps.indexOf(key) !== -1 && Object(consent)[key]) {
```

Comment:
is a need to use `Object(consent)[key]` instead of `consent[key]` ?

Reply:
if nullProtectedProps will be the map: {credentialId: true, ... } than this code would be simpler and there will be no need to scan the whole array using indexOf

Reply:
Consent, being an interface, doesn't allow normal key indexing. Rest of your suggestions have been addressed in a recent commit.